packages/runtime/container-runtime/src/dataStore.ts

```diff
@@ -202,3 +205,38 @@ class DataStore implements IDataStore {
 		});
 	}
 }
+
+export class DataStore2 implements IDataStore2 {
+	private _dataStore?: IDataStore;
+	public get context(): IFluidDataStoreContextDetached {
+		return this.fluidDatastoreContext;
+	}
+	constructor(
+		private readonly fluidDatastoreContext: LocalDetachedFluidDataStoreContext,
+		private readonly internalId: string,
+		private readonly runtime: ContainerRuntime,
+		private readonly datastores: DataStores,
+		private readonly logger: ITelemetryLoggerExt,
+	) {}
+
+	public get dataStore(): IDataStore {
+		if (this._dataStore === undefined) {
+			this._dataStore = channelToDataStore(
+				this.fluidDatastoreContext.channelValue,
+				this.internalId,
+				this.runtime,
+				this.datastores,
+				this.logger,
+			);
+		}
+		return this._dataStore;
+	}
+
+	public get entryPoint(): IFluidHandle<FluidObject> {
```

Comment:
It's not a good design for this getter to fail if some unrelated operation has not happened yet. That too on one of the parameters passed to this class. It might work if this did that operation itself and then returned the entry point.

Reply:
How about something like this: Remove `attachRuntime` from `IFluidDataStoreContextDetached` and add it to `IDataStore2`. And we add documentation that says this function is to be called before you can do anything meaningful like get entry point and trySetAlias, etc.
It's not great but its not necessarily worse that what we have today, i.e., created detached and then calling attachRuntime.

Reply:
Yeah, we can do that